### PR TITLE
Search: Debounce location update

### LINKED ISSUE #35010: Search: Do not update query in URL for every keystroke
Search feels a bit sluggish when typing, my suspicion is that it's because we update the url onChange and this triggers a full route re-render (incl sidemenu). Think we should change this to onBlur instead (ie update URL). This might improve performance.

### PR DESCRIPTION
**Pull request by Clarity-89**
**What this PR does / why we need it**:

Debounce updating location on search query change. 

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/35010

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/features/search/hooks/useSearchQuery.ts [hunk 1]
```diff
@@ -12,7 +14,8 @@ import {
 } from '../reducers/actionTypes';
 import { DashboardQuery, SearchLayout } from '../types';
 import { hasFilters, parseRouteParams } from '../utils';
-import { locationService } from '@grafana/runtime';
+
+const updateLocation = debounce((query) => locationService.partial(query), 300);
```
Comment (dprokop) [inline public/app/features/search/hooks/useSearchQuery.ts:18]: Any particular reason for debouncing? Can't we just use `onBlur`?

Comment (torkelo) [reply public/app/features/search/hooks/useSearchQuery.ts:18]: I think debounce might be better, onBlur will require separating state from url (right no url -> search query state -> do search)

**Review by torkelo — APPROVED** [review]

**Review by kaydelaney — APPROVED** [review]

File: public/app/features/search/hooks/useSearchQuery.ts (hunk 1 above)
Comment (dprokop) [reply public/app/features/search/hooks/useSearchQuery.ts:18]: oki doki, makes sense! Thanks @Clarity-89

**Comment by torkelo** [comment]
after using this for a while I think this was not a goog enough solution, search still feels super slow due to the serach input box not showing what I am typing until I stop

**Pull request merged**